validator/submain.py

```diff
@@ -46,49 +37,50 @@ def prepare_package(err, path, expectation=0, for_appversions=None,
     timeout is the number of seconds before validation is aborted.
     If timeout is -1 then no timeout checking code will run.
     """
-    # Test that the package actually exists. I consider this Tier 0
-    # since we may not even be dealing with a real file.
-    if err and not os.path.isfile(path):
-        err.error(("main",
-                   "prepare_package",
-                   "not_found"),
-                  "The package could not be found")
-        return
 
-    # Pop the package extension.
-    package_extension = os.path.splitext(path)[1]
-    package_extension = package_extension.lower()
-
-    if package_extension == ".xml":
-        return test_search(err, path, expectation)
-    elif expectation == PACKAGE_WEBAPP:
-        raise NotImplementedError(
-            "Webapps are not supported in amo-validator.")
-
-    # Test that the package is an XPI.
-    if package_extension not in (".xpi", ".jar"):
-        if err:
-            err.error(("main",
-                       "prepare_package",
-                       "unrecognized"),
-                      "The package is not of a recognized type.")
-        return False
+    package = None
+    try:
```

Comment:
The `try` doesn't need to enclose those following lines. I think it could be moved down to line 60